src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -61,15 +62,24 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
       case Failure(ex:ParsingException) => Future(RequestCompleteWithErrorReport(BadRequest, "Invalid json supplied", ex))
       case Failure(e) => Future(RequestCompleteWithErrorReport(BadRequest, BadRequest.defaultMessage, e))
       case Success(userAttrs) =>
-        rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
-          // verify owner on workspace
-          if (!workspaceResponse.accessLevel.contains("OWNER")) {
-            Future(RequestCompleteWithErrorReport(Forbidden, "must be an owner"))
-          } else {
-            // this is technically vulnerable to a race condition in which the workspace attributes have changed
-            // between the time we retrieved them and here, where we update them.
-            val allOperations = generateAttributeOperations(workspaceResponse.workspace.get.attributes, userAttrs)
-            rawlsDAO.patchWorkspaceAttributes(ns, name, allOperations) map (RequestComplete(_))
+        val validationResult = Try( schemaValidate(attrsJsonString) )
+        validationResult match {
+          case Failure(ve: ValidationException) =>
+            val errorReports = ve.getCausingExceptions.map{v => ErrorReport(v.getMessage)}
+            val userMsg = if (errorReports.size > 1)
+              ve.getMessage + ". See causes for details."
+            else
+              ve.getMessage
+            Future(RequestCompleteWithErrorReport(BadRequest, userMsg, errorReports))
+          case Failure(e) =>
+            Future(RequestCompleteWithErrorReport(BadRequest, BadRequest.defaultMessage, e))
+          case Success(x) => {
+            rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
+              // this is technically vulnerable to a race condition in which the workspace attributes have changed
+              // between the time we retrieved them and here, where we update them.
```

Comment:
From what I can grep, a second update to attributes will also result in a call to this actor method. If that's the case, what you'll get is two update actors trying to post to ES at roughly the same time so it's a coin toss as two which one is the last one. I think this will be true of *any* actor-based implementation.